server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -923,42 +924,47 @@ public PrimaryDataStoreInfo createPool(CreateStoragePoolCmd cmd) throws Resource
         return (PrimaryDataStoreInfo)_dataStoreMgr.getDataStore(store.getId(), DataStoreRole.Primary);
     }
 
-    private Map<String,String> extractUriParamsAsMap(String url){
+    protected Map<String,String> extractUriParamsAsMap(String url, boolean managed) {
         Map<String,String> uriParams = new HashMap<>();
         UriUtils.UriInfo uriInfo = UriUtils.getUriInfo(url);
```

Comment:
@shwstppr, it will fail here if the URL is not valid. Also, the Storpool plugin doesn't use the managed tag

Reply:
May check the first characters of url instead of the managed parameter

Reply:
@slavkap @weizhouapache thanks for the pointers. I'll make the change